docker/Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM quinoacomputing/buildenv:gentoo 
+
+ARG COVERAGE
+ARG CC
+ARG CXX
+ARG CXXFLAGS
+
+#for coverage
+ENV CI true
+ENV TRAVIS true
+ARG TRAVIS_BRANCH
+ARG TRAVIS_JOB_NUMBER
+ARG TRAVIS_PULL_REQUEST 
+ARG TRAVIS_JOB_ID
+ARG TRAVIS_TAG
+ARG TRAVIS_REPO_SLUG
+ARG TRAVIS_COMMIT
+
+RUN rm -rf /home/quinoa/.ccache
+COPY quinoa /home/quinoa/quinoa
+COPY ccache/ /home/quinoa/.ccache
+USER root
+RUN chown -R quinoa:quinoa /home/quinoa/quinoa /home/quinoa/.ccache
+USER quinoa
+
+WORKDIR /home/quinoa/quinoa
+RUN mkdir build
+
+WORKDIR build
+RUN ccache -z
+RUN cmake ${COVERAGE:+-DCODE_COVERAGE=ON} -DLCOV=/bin/true -DGENHTML=/bin/true ../src
+RUN make VERBOSE=1 -j2
+RUN ccache -s
+RUN ../script/run_tests.sh
+RUN make install DESTDIR=${PWD}
+RUN cd .. && if [ ${COVERAGE} ]; then \
+  if [ ${CC} = clang ]; then \
+    $HOME/.local/bin/codecov --gcov-exec "llvm-cov gcov"; \
```

Comment:
@jbakosi: that is where we would add the exclusion option. Or we put this command inside cmake.